src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/EndpointsServerPlugin.java

```diff
@@ -75,59 +79,83 @@ public void execute(Project project) {
+              genDiscoveryDocs.setDiscoveryDocDir(extension.getDiscoveryDocDir());
+              genDiscoveryDocs.setServiceClasses(extension.getServiceClasses());
+              genDiscoveryDocs.setWebAppDir(
+                  project.getConvention().getPlugin(WarPluginConvention.class).getWebAppDir());
+            }
+          });
+        }
+      });
+  }
+
+  private void createGenerateOpenApiDocsTask() {
+    project.getTasks().create(GENERATE_OPENAPI_DOC_TASK, GenerateOpenApiDocsTask.class,
+      new Action<GenerateOpenApiDocsTask>() {
+        @Override
+        public void execute(final GenerateOpenApiDocsTask genOpenApiDocs) {
+          genOpenApiDocs.setDescription("Generate endpoints openapi documents");
+          genOpenApiDocs.setGroup(APP_ENGINE_ENDPOINTS);
+          genOpenApiDocs.dependsOn(JavaPlugin.CLASSES_TASK_NAME);
+
+          project.afterEvaluate(new Action<Project>() {
+            @Override
+            public void execute(Project project) {
+              File classesDir = project.getConvention().getPlugin(JavaPluginConvention.class)
+                      .getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME).getOutput()
+                      .getClassesDir();
+              genOpenApiDocs.setClassesDir(classesDir);
+              genOpenApiDocs.setOpenApiDocDir(extension.getOpenApiDocDir());
+              genOpenApiDocs.setServiceClasses(extension.getServiceClasses());
+              genOpenApiDocs.setWebAppDir(
+                      project.getConvention().getPlugin(WarPluginConvention.class).getWebAppDir());
+            }
+          });
+        }
+      });
   }
 
   private void createGenerateClientLibsTask() {
     project.getTasks().create(GENERATE_CLINT_LIBS_TASK, GenerateClientLibsTask.class,
-        new Action<GenerateClientLibsTask>() {
-          @Override
-          public void execute(final GenerateClientLibsTask genClientLibs) {
-            genClientLibs.setDescription("Generate endpoints client libraries");
-            genClientLibs.setGroup(APP_ENGINE_ENDPOINTS);
-            genClientLibs.dependsOn(JavaPlugin.CLASSES_TASK_NAME);
-
-            project.afterEvaluate(new Action<Project>() {
-              @Override
-              public void execute(Project project) {
-                File classesDir = project.getConvention().getPlugin(JavaPluginConvention.class)
-                    .getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME).getOutput()
-                    .getClassesDir();
-                genClientLibs.setClassesDir(classesDir);
-                genClientLibs.setClientLibDir(extension.getClientLibDir());
-                genClientLibs.setServiceClasses(extension.getServiceClasses());
-                genClientLibs.setWebAppDir(
-                    project.getConvention().getPlugin(WarPluginConvention.class).getWebAppDir());
-              }
-            });
-          }
-        });
+      new Action<GenerateClientLibsTask>() {
```

Comment:
please undo the spacing changes here, they were correct before (offset by 2x for line continuation)